prometheus/testutil/promlint/promlint.go

```diff
@@ -110,276 +99,25 @@ func (l *Linter) Lint() ([]Problem, error) {
-	}
-
-	return problems
-}
-
-// lintMetricUnits detects issues with metric unit names.
-func lintMetricUnits(mf *dto.MetricFamily) []Problem {
-	var problems []Problem
-
-	unit, base, ok := metricUnits(*mf.Name)
-	if !ok {
-		// No known units detected.
-		return nil
-	}
-
-	// Unit is already a base unit.
-	if unit == base {
-		return nil
-	}
-
-	problems = append(problems, newProblem(mf, fmt.Sprintf("use base unit %q instead of %q", base, unit)))
-
-	return problems
-}
-
-// lintCounter detects issues specific to counters, as well as patterns that should
-// only be used with counters.
-func lintCounter(mf *dto.MetricFamily) []Problem {
-	var problems []Problem
-
-	isCounter := mf.GetType() == dto.MetricType_COUNTER
-	isUntyped := mf.GetType() == dto.MetricType_UNTYPED
-	hasTotalSuffix := strings.HasSuffix(mf.GetName(), "_total")
-
-	switch {
-	case isCounter && !hasTotalSuffix:
-		problems = append(problems, newProblem(mf, `counter metrics should have "_total" suffix`))
-	case !isUntyped && !isCounter && hasTotalSuffix:
-		problems = append(problems, newProblem(mf, `non-counter metrics should not have "_total" suffix`))
-	}
-
-	return problems
-}
-
-// lintHistogramSummaryReserved detects when other types of metrics use names or labels
-// reserved for use by histograms and/or summaries.
-func lintHistogramSummaryReserved(mf *dto.MetricFamily) []Problem {
-	// These rules do not apply to untyped metrics.
-	t := mf.GetType()
-	if t == dto.MetricType_UNTYPED {
-		return nil
-	}
-
-	var problems []Problem
-
-	isHistogram := t == dto.MetricType_HISTOGRAM
-	isSummary := t == dto.MetricType_SUMMARY
-
-	n := mf.GetName()
-
-	if !isHistogram && strings.HasSuffix(n, "_bucket") {
-		problems = append(problems, newProblem(mf, `non-histogram metrics should not have "_bucket" suffix`))
-	}
-	if !isHistogram && !isSummary && strings.HasSuffix(n, "_count") {
-		problems = append(problems, newProblem(mf, `non-histogram and non-summary metrics should not have "_count" suffix`))
-	}
-	if !isHistogram && !isSummary && strings.HasSuffix(n, "_sum") {
-		problems = append(problems, newProblem(mf, `non-histogram and non-summary metrics should not have "_sum" suffix`))
-	}
-
-	for _, m := range mf.GetMetric() {
-		for _, l := range m.GetLabel() {
-			ln := l.GetName()
-
-			if !isHistogram && ln == "le" {
-				problems = append(problems, newProblem(mf, `non-histogram metrics should not have "le" label`))
-			}
-			if !isSummary && ln == "quantile" {
-				problems = append(problems, newProblem(mf, `non-summary metrics should not have "quantile" label`))
-			}
-		}
-	}
-
-	return problems
-}
-
-// lintMetricTypeInName detects when metric types are included in the metric name.
-func lintMetricTypeInName(mf *dto.MetricFamily) []Problem {
-	var problems []Problem
-	n := strings.ToLower(mf.GetName())
-
-	for i, t := range dto.MetricType_name {
-		if i == int32(dto.MetricType_UNTYPED) {
-			continue
+	for _, fn := range defaultValidations {
```

Comment:
Why, why not merging those slices (default and custom) and not repeating the code?

Not a blocker, that's the only nit I would improve, merging anyway.